src/battle_util.c

```diff
@@ -7830,3 +7830,98 @@ u8 GetBattleMoveSplit(u32 moveId)
     else
         return SPLIT_SPECIAL;
 }
+
+// sort an array of battlers by speed
+//  useful for effects like pickpocket, eject button, red card, dancer
+void SortBattlersBySpeed(u8 *battlers, bool8 slowToFast)
+{
+	int i, j, key, keyBank;
+	u16 speeds[4] = {0};
+    
+	for (i = 0; i < gBattlersCount; i++)
+		speeds[i] = GetBattlerTotalSpeedStat(battlers[i]);
+
+	for (i = 1; i < gBattlersCount; i++)
+	{
+		keyBank = battlers[i];
+		key = speeds[i];
+		j = i - 1;
+
+		if (slowToFast)
+		{
+			while (j >= 0 && speeds[j] > key)
+			{
+				battlers[j + 1] = battlers[j];
+				speeds[j + 1] = speeds[j];
+				j = j - 1;
+			}
+		}
+		else
+		{
+			while (j >= 0 && speeds[j] < key)
+			{
+				battlers[j + 1] = battlers[j];
+				speeds[j + 1] = speeds[j];
+				j = j - 1;
+			}
+		}
+
+		battlers[j + 1] = keyBank;
+		speeds[j + 1] = key;
+	}
+}
```

Comment:
I feel like the terms `key` and `bank` probably need to be renamed to be more clear for what they're supposed to be. Bank especially I believe was an old term used in the IDBs for FR and E which was renamed to `battler` in the Decomps.